### DISCUSSION TITLE
[Feature] Explore Kotlin Multiplatform for desktop app

### DISCUSSION THREAD

**Discussion by nomorsug** [TechbeeAT/jtxBoard#920] [Ideas]
Currently, there isn't any good desktop CalDAV implementation. [Kotlin Multiplatform](https://kotlinlang.org/docs/multiplatform.html) could allow porting jtxBoard to desktop/iOS without having to rewrite the app from scratch.

https://github.com/simplex-chat/simplex-chat is a good example of Android app ported to desktop from a single codebase (this is still WIP but proves its feasible)

**Comment by patrickunterwegs** [comment]
For sure this is an interesting topic. I would really like to dig deeper into KMM, but for now I had to take a new full-time job and I'm more limited in time for the next months, so this won't happen any time soon. I'll keep this topic in discussions though.

**Reply by miseenplace** [reply]
Any chance you could estimate the time on this? I would be happy to open a bounty on this (on whatever platform you think best) to fund this work.

Having this app on iOS (with widgets) would be great. Desktop would be a bonus, though the UI optimization adjustments / beyond scales existing mobile / tablet views would be a later point to tackle.

The separate costs of developer submission to iOS app store should also be factored into later funding objective.

Not sure how this would affect the app storage / data access model, only took a cursory examination if the multi-platform port documentation. There is separate documentation for UI/UX and backend.
